doc/reference/kernel/other/float.rst

```diff
@@ -193,6 +193,19 @@ when there is no need to do so.
 When the thread again needs to use the floating point registers it can re-tag
 itself as an FPU user or SSE user by calling :cpp:func:`k_float_enable()`.
 
+
+RISCV architecture
+------------------
+
+On the RISCV architecture, the kernel simply treats all thread as an FPU user
```

Comment:
Well, that certainly simplifies things.
How about this to fix some misspellings and for clarity:
```
On the RISCV architecture, the kernel treats all threads as an FPU user
if :option:`CONFIG_FLOAT` and :option:`CONFIG_FP_SHARING` are
selected. This may lead to performance problems because the FP
registers will be saved and restored on every thread context switch.

In an optimized implementation, the kernel should only save and restore
FP registers on a thread context switch if the thread is tagged as an FPU
user. An extra 132 bytes (4 x 32 single precision + fscr)
or 260 bytes (8 x 32 double precision + fscr) of stack space is
required to load and store floating point registers.  
```